app/models/event.rb

```diff
@@ -87,6 +87,9 @@ class Event < ApplicationRecord
       # TODO: SOLR has a LatLonType to do geospatial searching. Have a look at that
       #       location :latitutde
       #       location :longitude
+      string :collections, multiple: true do
+        collections.where(public: true).map(&:title)
```

Comment:
`.pluck(:title)` would be faster

Reply:
only if the collections are not already in memory (which they could be). But I'll change it to pluck, and a performance analysis can be done at a later stage.